lib/jazzy/grouper.rb

```diff
@@ -69,7 +69,42 @@ def self.group_custom_categories(docs, doc_index)
         children.each.with_index { |child, i| child.nav_order = i }
         make_group(children, category['name'], '')
       end
-      [group.compact, docs]
+
+      globals_grouped_by_file = self.group_globals_by_file(docs, doc_index)
+      [group.compact, docs + globals_grouped_by_file]
+    end
+
+    # Group root level Constants and Functions by the file they're declared in.
+    def self.group_globals_by_file(docs, doc_index)
+      types_to_files_to_symbols = {}
+      # Group symbols by file and type.
+      doc_index.each do |index_item|
+        if index_item.parent_in_code.nil? && (index_item.type.name == "Constant" || index_item.type.name == "Function")
```

Comment:
You can just use `docs` instead of `doc_index` if you just want top-level items, then won't need the `parent_in_code` check and will be more efficient.